modules/currency-l1/src/main/scala/org/tessellation/currency/l1/CurrencyL1App.scala

```diff
@@ -85,10 +86,14 @@ abstract class CurrencyL1App(
           seedlist
         )
       dagP2PClient = DAGP2PClient
-        .make[IO](sdkP2PClient, sdkResources.client, currencyPathPrefix = "currency")
+        .make[IO](sdkP2PClient, sdkResources.client, currencyPathPrefix = "currency", method.globalL0Peer)
+      selectedL0Peer <- storages.l0Cluster.getPeers.map { l0Peers =>
+        Random.shuffle(l0Peers.toNonEmptyList.toList).head
```

Comment:
duplication for peer select logic

Reply:
Moved the peer selection logic into `RandomPeerSelect` which is passed all the `L0Peer`s.